test/vcf/record_test.cpp

```diff
@@ -510,5 +536,98 @@ namespace ebi
                                 source}),
                             vcf::IdBodyError*);
         }
+
+        SECTION("Conflicting data line values and meta header definition")
+        {
+            CHECK_THROWS_AS( (vcf::Record{
+                            1,
+                            "chr1",
+                            123456,
+                            { "id123", "id456" },
+                            "A",
+                            { "AC", "AT" },
+                            1.0,
+                            { "PASS" },
+                            { {"InfoTag", "1.89"}, { "AF", "0.5,0.3"} },
+                            { "GT", "DP" },
+                            { "0|1" },
+                            source}),
+                        vcf::InfoBodyError*);
+
+            CHECK_THROWS_AS( (vcf::Record{
+                            1,
+                            "chr1",
+                            123456,
+                            { "id123", "id456" },
+                            "A",
+                            { "AC", "AT" },
+                            1.0,
+                            { "PASS" },
+                            { {"InfoTag", "1,2,3"}, { "AF", "0.5,0.3"} },
+                            { "GT", "DP" },
+                            { "0|1" },
+                            source}),
+                        vcf::InfoBodyError*);
+
+            CHECK_THROWS_AS( (vcf::Record{
+                            1,
+                            "chr1",
+                            123456,
+                            { "id123", "id456" },
+                            "A",
+                            { "AC", "AT" },
+                            1.0,
+                            { "PASS" },
+                            { {"AN", "12"}, { "AF", "0.5,0.3"} },
+                            { "GT", "FormatTag" },
+                            { "0|1:ta,gs" },
+                            source}),
+                        vcf::SamplesFieldBodyError*);
+
+            CHECK_THROWS_AS( (vcf::Record{
+                            1,
+                            "chr1",
+                            123456,
+                            { "id123", "id456" },
+                            "A",
+                            { "AC", "AT" },
+                            1.0,
+                            { "PASS" },
+                            { {"AN", "12"}, { "AF", "0.5,0.3"} },
+                            { "GT", "FormatTag" },
+                            { "0|1:1.5" },
+                            source}),
+                        vcf::SamplesFieldBodyError*);
```

Comment:
Check for any normal(non-predefined) tag. The format tag does not match the meta definition Number here.